toolkit/tools/internal/packagerepo/repocloner/rpmrepocloner/rpmrepocloner.go

```diff
@@ -292,20 +320,20 @@ func (r *RpmRepoCloner) initializeRepoDefinitions(disableDefaultRepos bool, repo
 	// Add each previously existing repofile to the end of the new file, then delete the original.
 	// We want to try our custom mounted repos before reaching out to the upstream servers.
 	// By default, chroot ships with PMC repositories specified in mariner-repos rpm.
-	// If `disableDefaultRepos` flag is turned on, we only remove these existing files from the
-	// chroot repo directory. tdnf will not reach out to the default PMC repositories for lookups.
 	for _, originalRepoFilePath := range existingRepoFiles {
-		if !disableDefaultRepos {
-			err = appendRepoFile(originalRepoFilePath, dstFile)
-			if err != nil {
-				return
-			}
-		} else {
-			logger.Log.Debugf("Disabling repositories listed in %s", originalRepoFilePath)
+		repoIDs, err := readRepoIDs(originalRepoFilePath)
+		if err != nil {
+			return err
+		}
+		r.defaultMarinerRepoIDs = append(r.defaultMarinerRepoIDs, repoIDs...)
```

Comment:
I think we are poluting the default repos here with any custom repo files the user passes in.

Reply:
When the call to `disabledDefaultMarinerReposArgs()` is made inside `SetEnabledRepos()` I think it will also turn off the custom repo files.

Reply:
Fortunately, we're not. The default repos are read from inside the chroot's directory. The user-defined ones are never copied into the chroot and we only parse them from the paths provided by the user.

Reply:
Allright, maybe add a comment on the order of operations but seems reasonable.